File: Cpp/1893_Check_if_All_the_Integers_in_a_Range_Are_Covered.cpp
class Solution {
public:
    bool isCovered(vector<vector<int>>& ranges, int left, int right) {
        vector<int> dp(52, 0);
        for(int i = 0; i < ranges.size(); ++i) {            
            ++dp[ranges[i][0]];
            --dp[ranges[i][1] + 1];
        }

        for(int i = 1; i < 52; ++i) {
            dp[i] += dp [i - 1];
        }

        for(int i = left; i <= right; ++i) {
            if(dp[i] == 0) return false;
        }

        return true;
    }
};

File: Kotlin/1893_Check_if_All_the_Integers_in_a_Range_Are_Covered.kt
class Solution {
    fun isCovered(ranges: Array<IntArray>, left: Int, right: Int): Boolean {
        val dp = IntArray(52)
        for (range in ranges) {
            ++dp[range[0]]
            --dp[range[1] + 1]
        }

        for(i in 1 until 52) {
            dp[i] += dp[i - 1]
        }

        for(i in left..right) {
            if(dp[i] == 0) return false
        }

        return true
    }
}